File: go/Regex.go
import "regexp"

r := regexp.MustCompile(`expression`)
matches := r.FindAllString("...", -1) //len(matches) ..

r = regexp.MustCompile(`<hello>(.*)</world>`)
k := r.FindAllStringSubmatch("<hello>oscar</world>", -1)
// k[0][0] is full match
// k[0][1] first group (oscar)

File: ts/regex.ts
let reg = new RegExp(/h.* /g);
let text = "hello world";

let result = reg.exec(text);
console.log(result);

// result != null => match was found.
// result[0] - the matched string ("hello "")

// result == null => no match found.

// ----------------------------------------------------------

const regex = /Note (\d)/;
const result = regex.exec('Note 1');
// result[1] - first group

// ----------------------------------------------------------
const regex = /[A-Z]/g;
const s = "OsCaR";
let result: RegExpExecArray;
while ((result = regex.exec(s)) != null) {
  console.log(result[0]);
}

// O C R